src/main/java/com/google/sps/servlets/RecipeServlet.java

```
@@ -37,4 +58,39 @@ public void doPut(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doDelete(HttpServletRequest request, HttpServletResponse response)
       throws IOException {}
+
+  private void getRecipeList(HttpServletResponse response) throws IOException {
+    Query query = DBReferences.recipes;
+    ApiFuture<QuerySnapshot> querySnapshot = query.get();
+    ArrayList<Object> recipeList = new ArrayList<>();
+
+    try {
+      for (DocumentSnapshot document : querySnapshot.get().getDocuments())
+        recipeList.add(document.getData());
+
+      String json = gson.toJson(recipeList);
+      response.setContentType("application/json;");
+      response.getWriter().println(json);
+    } catch (Exception e) {
+      System.out.println("Recipe GET threw exception: " + e);
+    }
+  }
+
+  private void getDetailedRecipe(String recipeID, HttpServletResponse response) throws IOException {
+    DocumentReference recipeRef = DBReferences.recipes.document(recipeID);
+    ApiFuture<DocumentSnapshot> future = recipeRef.get();
+
+    try {
+        DocumentSnapshot document = future.get();
+        if (document.exists()) {
+            String json = gson.toJson(document.getData());
+            response.setContentType("application/json;");
+            response.getWriter().println(json);
+        } else {
+            System.out.println("No such document!");
+        }
+    } catch (Exception e) {
```

Avoid catch-all exceptions - catch only what'd expect it to throw.